Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -658,6 +659,157 @@ class InAppMessageViewTest: IntegrationTest {
         // If url is valid, check if `handleDeepLink` method is called
         XCTAssertFalse(deeplinkUtilMock.handleDeepLinkCalled)
     }
+
+    // MARK: - Track open
+
+    @MainActor
+    func test_onInlineMessageShown_expectTrackOpenedMetric_expectGlobalMessageShownListener() async {
+        let inlineView = await showInlineMessageForMetrics()
+
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        // Check if messageShown is called
+        XCTAssertTrue(eventListenerMock.messageShownCalled)
+        XCTAssertEqual(eventListenerMock.messageShownCallsCount, 1)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 1)
+    }
+
+    @MainActor
+    func test_givenMultipleInlineMessageInQueue_BothDisplayedAndOneDismissed_expectTrackMultipleMessageShownListenerCalls() async {
+        messagingInAppImplementation.setEventListener(eventListenerMock)
+        messagingInAppImplementation.setEventBusHandler(eventBusHandlerMock)
+
+        let givenInlineMessage1 = Message.randomInline
+        let givenInlineMessage2 = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage1, givenInlineMessage2]
+
+        let inlineView1 = InAppMessageView(elementId: givenInlineMessage1.elementId!)
+        let inlineView2 = InAppMessageView(elementId: givenInlineMessage2.elementId!)
+
+        // Render only inlineView1
+        await onDoneRenderingInAppMessage(givenInlineMessage1, insideOfInlineView: inlineView1)
+
+        // inlineView1 is only visible
+        XCTAssertTrue(isInlineViewVisible(inlineView1))
+        XCTAssertFalse(isInlineViewVisible(inlineView2))
+
+        // Check if messageShown is called
+        XCTAssertTrue(eventListenerMock.messageShownCalled)
+        XCTAssertEqual(eventListenerMock.messageShownCallsCount, 1)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 1)
+
+        // Dismiss inlineView1 and show inlineView2
+        await onCloseActionButtonPressed(onInlineView: inlineView1)
+        await onDoneRenderingInAppMessage(givenInlineMessage2, insideOfInlineView: inlineView2)
+
+        // Check messageShownCalled called again
+        XCTAssertTrue(eventListenerMock.messageShownCalled)
+        XCTAssertEqual(eventListenerMock.messageShownCallsCount, 2)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 2)
+    }
+
+    // MARK: - Track click
+
+    @MainActor
+    func test_onInlineMessageShown_givenDelegateNotSet_onSingleButtonTap_expectTrackClickedMetric_expectGlobalMessageActionTakenCallback() async {
```

Comment:
I think this test should be broken into multiple test functions. 

* given button clicked - expect track metric
* given delegate not set, given button click, expect global event listener called. <-- I consider this test case out of scope for this PR.